### PR TITLE
Features/uc gr uow

### PR DESCRIPTION
**Pull request by SvyatoslavFedynyak**
I left comments to know what to do if something will be wrong. In final version I'll delete em.

### CODE REVIEW

File: Task5/Project/TaxiDriver/Context/DriverContext.cs [hunk 1]
```diff
@@ -2,17 +2,100 @@
...
+        DbSet<Route> routes;
+        DbSet<Score> scores;
+
         // CONSTRUCTORS
+        /// <summary>
+        /// Basic constructor for context class
+        /// </summary>
         public DriverContext()
-            : base("TaxiDtiverDB")  {  }
+            : base("TaxiDriverDB")  {  }
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Context/DriverContext.cs]: yeah, thanks

File: Task5/Project/TaxiDriver/Context/DriverContext.cs [hunk 2]
```diff
@@ -2,17 +2,100 @@
...
+            {
+                return drivers;
+            }
+            set
+            {
+                drivers = value;
+            }
+        }
+        /// <summary>
+        /// Property that enable to interruct with clients data
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Context/DriverContext.cs]: *interact

File: Task5/Project/TaxiDriver/Context/DriverContext.cs [hunk 3]
```diff
@@ -2,17 +2,100 @@
...
+            }
+            set
+            {
+                clients = value;
+            }
+        }
+        /// <summary>
+        /// Property that enable to interruct with streets data
+        /// </summary>
+        /// <returns>streets data</returns>
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Context/DriverContext.cs]: Upprcase

File: Task5/Project/TaxiDriver/Services/GenericRepository.cs [hunk 4]
```diff
@@ -6,20 +6,39 @@
...
         }
         // PROPERTIES
+        /// <summary>
+        /// Property that enale to interruct with count of entities in data base
+        /// </summary>
+        /// <returns>Count of entities</returns>
         public int Count => dbSet.Count();
         // METHODS
+        /// <summary>
+        /// Method that data from data base
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Services/GenericRepository.cs]: What it does with data?

File: Task5/Project/TaxiDriver/Services/GenericRepository.cs [hunk 5]
```diff
@@ -43,18 +62,35 @@ public GenericRepository(Context.DriverContext context)
...
+        /// <summary>
+        /// Deletes object by id
+        /// </summary>
+        /// <param name="id">Objects id</param>
         public virtual void Delete(object id)
         {
             Delete(dbSet.Find(id));
         }
+        /// <summary>
+        /// Deletes preset entity
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Services/GenericRepository.cs]: *present

File: Task5/Project/TaxiDriver/Services/UnitOfWork.cs [hunk 6]
```diff
@@ -3,85 +3,146 @@
...
     {
         // FIELDS
+        GenericRepository<Client> clientRepository;
+        GenericRepository<DriverInfo> driverInfoRepository;
+        GenericRepository<Route> routeRepository;
+        GenericRepository<Score> scoreRepository;
+        GenericRepository<Street> streetRepository;
+        Context.DriverContext db;
+
         private bool disposedValue = false; // To detect redundant calls
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Services/UnitOfWork.cs]: should be in constructor

File: Task5/Project/TaxiDriver/Services/UnitOfWork.cs [hunk 7]
```diff
@@ -3,85 +3,146 @@
...
-
+        /// <summary>
+        /// Save changes in data base
+        /// </summary>
         public void Save()
         {
-            throw new NotImplementedException();
+            clientRepository.context.SaveChanges();
+            driverInfoRepository.context.SaveChanges();
+            routeRepository.context.SaveChanges();
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Services/UnitOfWork.cs]: just save DB. Why would save the same context multiple time?

File: Task5/Project/TaxiDriver/Services/UnitOfWork.cs [hunk 8]
```diff
@@ -3,85 +3,146 @@
...
         protected virtual void Dispose(bool disposing)
         {
             if (!disposedValue)
             {
                 if (disposing)
                 {
                     // TODO: dispose managed state (managed objects).
+                    clientRepository = null;
+                    driverInfoRepository = null;
+                    routeRepository = null;
```
Comment (iamprovidence) [inline Task5/Project/TaxiDriver/Services/UnitOfWork.cs]: should be only db.Dispose() here.

File: Task5/Project/TaxiDriver/Services/GenericRepository.cs (hunk 5 above)
Comment (SvyatoslavFedynyak) [reply Task5/Project/TaxiDriver/Services/GenericRepository.cs]: > 
> *present
>
No, preset

Comment (iamprovidence) [reply Task5/Project/TaxiDriver/Services/GenericRepository.cs]: ok

**Pull request merged**